snapshot/git/gitdb/bundlestore.go

```diff
@@ -96,6 +96,14 @@ func (b *bundlestoreBackend) uploadLocalSnapshot(s *localSnapshot, db *DB) (sn s
 
 			mergeBase, err := db.dataRepo.RunSha("merge-base", streamHead, commitSha)
 
+			if mergeBase == commitSha {
```

Comment:
When would this happen? I may be misunderstanding, but only if the user committed something to master and then later decided to revisit that revision? And this works because only SQ can merge to master?